pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -75,6 +84,56 @@ func DiffList(olds, news []Manifest, opts ...diff.Option) (*DiffListResult, erro
 	return cr, nil
 }
 
+// If at least one of old and new is nil or not secret, then return new, nil
+func normalizeNewSecret(old, new *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	if old == nil || new == nil {
+		return new, nil
+	}
+
+	gvkOld := old.GroupVersionKind()
+	gvkNew := new.GroupVersionKind()
+
+	if gvkOld.Kind != "Secret" || gvkNew.Kind != "Secret" {
+		return new, nil
+	}
```

Comment:
Let's use this helper to check.
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/cloudprovider/kubernetes/resourcekey.go#L176